modules/document_repository/js/document_repository_helper.js

```diff
@@ -145,6 +201,72 @@ function postEdit(id) {
   });
 }
 
+function editCategoryModal() {
+  "use strict";
+
+  var id = this.id;
+  $("#editCategoryModal").modal();
+
+  $.ajax({
+    type: "GET",
+    url: loris.BaseURL + "/document_repository/ajax/getCategory.php",
+    data: {id: id},
+    async: false,
```

Comment:
does this need to be synchronous? would it be feasible to use a promise here?

Reply:
Definitely should be feasible I assume. Will do. 👍